sklearn/tree/tree.py

```diff
@@ -240,8 +256,17 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                              "number of samples=%d" % (len(y), n_samples))
         if self.min_samples_split <= 0:
             raise ValueError("min_samples_split must be greater than zero.")
+        if not 1 < min_samples_split <= n_samples:
+            raise ValueError("min_samples_split must in [2, n_samples]")
```

Comment:
It's different, the first one test both the percentage version and non percentage version, while the second also tests the int version.